tiledb/CMakeLists.txt

```diff
@@ -561,6 +623,97 @@ if (CURL_APPEND_LIBS)
     ${CURL_APPEND_LIBS})
 endif()
 
+############################################################
+# capnproto file generation
+############################################################
+
+if(TILEDB_SERIALIZATION)
+  if(TILEDB_CAPNP_EP_BUILT)
+    set(TILEDB_CAPNPROTOEXEPATH "${TILEDB_CAPNPEXEC_PATH}")
+  else()
+    find_package(CapnProto 0.6.1 #*not* EXACT since windows/*n*x using 0.8.0, 0.6.1 respectively
+      PATHS  ${TILEDB_EP_INSTALL_PREFIX} ${TILEDB_DEPS_NO_DEFAULT_PATH}
+    )
+    set(TILEDB_CAPNPROTOEXEPATH "${CAPNP_EXECUTABLE}")
+  endif()
+
+###TBD: REMOVEME:
+  message(STATUS "TILEDB_CAPNPROTOEXEPATH: ${TILEDB_CAPNPROTOEXEPATH}")
+
+  if(0)
+  set(TILEDB_CAPNPROTOGENCMDARGS " compile -I ${TILEDB_EP_BASE}/install/include -oc++ tiledb-rest.capnp")
+  set(TILEDB_CAPNPROTOGENCMD "${TILEDB_CAPNPROTOEXEPATH} ${TILEDB_CAPNPROTOGENCMDARGS}")
+  endif()
+
+  if(1)
+  #force (re-)gen to run each time so can alternate msvc/wsl builds without removing *-capnp.<h,c++>
+  if(WIN32)
```

Comment:
Does the `capnp_generate_cpp` macro do what we need, in order to simplify this?https://github.com/capnproto/capnproto/blob/3e20359d019c5841a8eaffb8099f45c54e9fae48/c%2B%2B/cmake/CapnProtoMacros.cmake

(could avoid shelling out to separate scripts, etc.)

Reply:
(I would also be curious if the output is still different when using the upstream macro rather than separate update drivers)

Reply:
While did see capnp_generate_cpp hadn't considered usage, was working from tiledb wiki info for (re-)genning files.

Expect output to be same even if using capnp_generate_cpp, differences due to capnproto working around name collisions with ERROR/VOID definitions in windows.h so involves capnproto windows specific include of capnproto windows-sanity.h file.

Reply:
I'd be in favor of the upstream macro, seems like it'll be easier to maintain long term. We can also upgrade osx/linux to 0.8 if it simplifies the macro usage.

Reply:
(long, maybe this conversation needs to move elsewhere?)
summary: capnproto 0.8.0 requires c++14.  Neither current minimum gcc (think 4.8, but could not quickly find stated, just gcc supporting c++11 requirement) nor vs2015 support that.
Approximately current 'dev' (tried today), on windows, vs2015 build, 
some Hilbert-related unit tests fail with just 'basic' tiledb build configuration.
...
have recalled, realized while development in process (discussed only with @joe-maley )
another reason that came up to stay with *nix capnproto 0.6.1 is that 0.8.0 states requirement of c++14, 
which tiledb current minimum of gcc (4.8?) apparently does not fully/reliably support c++14.
Hence, may need to wait until gcc minimum supporting c++14 specified 
before being able to move to capnproto 0.8.0 on *nix builds.
IIRC correctly, the *nix builds I tried with 0.8.0 happened to have gcc 7.something in use.
MSVC 2017 sufficiently supports it.
(think somewhere we currently specify minimum msvc 2015, which does not support c++14, but also seem to recall encountering something other than this that would actually prohibit a successful build of tiledb with vs2015.
Offhand, cmake itself is not included with vs2015.)
Obtaining vanilla win32 cmake 3.3 from kitware, using vs2015,
configuring build withOUT serialization, tiledb core builds but with multiple warnings reporting 'previous versions of the compiler did not override when parameters only differed by const/volatile qualifiers' all against MemFileSystem::<member>s, 
and tiledb_unit builds, but (at least) some Hilbert tests are failing.
Building with serialization, capnproto configure states
'#error:  "You need Visual Studio 2017 or better to compile this code."'

Reply:
@ihnorton, regarding capnp_generate_cpp doing what we need...
I have failed to make it work within my current usage, 
tho' it will generate source if used 'just so'.
So I would like to stick with what's already in place, 
tho willing to try more if you can offer guidance to overcome my current issue, detail following.

My current usage has the source being (re-)generated on each build attempt,
the reason for this usage being so I can build from the same source tree in both a windows and *nix (wsl) context 
(tho' not with overlapping time intervals, i.e. only one of them building at a time).
Trying to script it's usage runs afoul at least for an add_library that is apparently part of the capnproto find_package support, as cmake reports add_library is not scriptable, and it is the finding of the package that also results in the capnp_generate_cpp() function/macro being available for use.

Reply:
Currently this generates the files in the source tree, so one way to solve that problem is to put the generated files in the build tree rather than the source tree.

IMO that preferable anyway because these files will not be checked in.

Reply:
as commented elsewhere, gen'd <.h,.c++> files going to individual build directories rather than source tree, tho' generation is still being done with custom scripts since necessary variables to use capnproto (capnp_generate_cpp) to generate to specific output directory are not available/functional thru v0.8.0